src/dmd/dtemplate.d

```diff
@@ -5808,6 +5803,35 @@ extern (C++) class TemplateInstance : ScopeDsymbol
     TemplateInstance tnext;     // non-first instantiated instances
     Module minst;               // the top module that instantiated this instance
 
+    ushort nest;                // for recursive pretty printing detection, 3 MSBs reserved for flags (below)
+    ubyte inuse;                // for recursive expansion detection
+    private enum Flag : uint { semantictiargsdone = 1u << 31, havetempdecl = 1u << 30, gagged = 1u << 29 }
+
+    final @safe @property pure nothrow @nogc
+    {
+        /// has semanticTiargs() been done?
+        bool semantictiargsdone() const { return (nest & Flag.semantictiargsdone) != 0; }
+        void semantictiargsdone(bool x)
+        {
+            if (x) nest |= Flag.semantictiargsdone;
```

Comment:
Good luck storing `1<<31` in an ushort ;-)

Reply:
Oi. Thanks! Fixed now so the flags are automatically chosen depending on `nest.sizeof`.